internal/tiltfile/updatesettings/update_settings.go

```diff
@@ -26,26 +27,57 @@ func (e Extension) OnStart(env *starkit.Environment) error {
 	return env.AddBuiltin("update_settings", e.updateSettings)
 }
 
-func (e Extension) updateSettings(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
-	var maxParallelUpdates int
+func (e *Extension) updateSettings(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var maxParallelUpdates, k8sUpsertTimeoutSecs starlark.Value
 	if err := starkit.UnpackArgs(thread, fn.Name(), args, kwargs,
-		"max_parallel_updates", &maxParallelUpdates); err != nil {
+		"max_parallel_updates?", &maxParallelUpdates,
+		"k8s_upsert_timeout_secs?", &k8sUpsertTimeoutSecs); err != nil {
 		return nil, err
 	}
 
-	if maxParallelUpdates < 1 {
+	mpu, mpuPassed, err := valueToInt(maxParallelUpdates)
+	if err != nil {
+		return nil, errors.Wrap(err, "update_settings: for parameter \"max_parallel_updates\"")
+	}
+	if mpuPassed && mpu < 1 {
 		return nil, fmt.Errorf("max number of parallel updates must be >= 1(got: %d)",
 			maxParallelUpdates)
 	}
 
-	err := starkit.SetState(thread, func(settings model.UpdateSettings) model.UpdateSettings {
-		settings = settings.WithMaxParallelUpdates(maxParallelUpdates)
+	kuts, kutsPassed, err := valueToInt(k8sUpsertTimeoutSecs)
+	if err != nil {
+		return nil, errors.Wrap(err, "update_settings: for parameter \"k8s_upsert_timeout_secs\"")
+	}
+	if kutsPassed && kuts < 1 {
+		return nil, fmt.Errorf("minimum k8s upsert timeout is 1s; got %ds",
+			k8sUpsertTimeoutSecs)
+	}
+
+	err = starkit.SetState(thread, func(settings model.UpdateSettings) model.UpdateSettings {
+		if mpuPassed {
+			settings = settings.WithMaxParallelUpdates(mpu)
+		}
+		if kutsPassed {
+			settings = settings.WithK8sUpsertTimeout(time.Duration(kuts) * time.Second)
+		}
 		return settings
 	})
 
 	return starlark.None, err
 }
 
+func valueToInt(v starlark.Value) (val int, wasPassed bool, err error) {
+	switch x := v.(type) {
+	case nil:
```

Comment:
might be good to include starlark.None here

Reply:
hm you mean if it's nil OR starlark.None, assume it's not set?